### PR TITLE
add missing fields to Deal, Source and User model

### PR DESCRIPTION
**Pull request by ppietr**
add missing fields to models:
Deal:
- last_stage_change_at
- last_stage_change_by_id
- last_activity_at
- estimated_close_date

Source:
- resource_type 

User:
- deleted_at

### CODE REVIEW

File: src/main/java/com/getbase/models/Deal.java [hunk 1]
```diff
@@ -32,6 +32,10 @@
   protected @JsonView(Views.ReadWrite.class) BigDecimal value;
   protected @JsonView(Views.ReadWrite.class) List<String> tags = new ArrayList<String>();
   protected @JsonView(Views.ReadWrite.class) Map<String, Object> customFields = new HashMap<String, Object>();
+  protected @JsonView(Views.ReadWrite.class) DateTime lastStageChangeAt;
```
Comment (michal-mally) [inline src/main/java/com/getbase/models/Deal.java]: This one should be`ReadOnly`.

File: src/main/java/com/getbase/models/Deal.java [hunk 2]
```diff
@@ -150,6 +170,14 @@ public void setCustomFields(Map<String, Object> customFields) {
     this.customFields = customFields;
   }
 
+  public void setLastStageChangeAt(DateTime lastStageChangeAt) {
```
Comment (michal-mally) [inline src/main/java/com/getbase/models/Deal.java]: There should be no setter for this field.

File: src/main/java/com/getbase/models/Source.java [hunk 3]
```diff
@@ -13,6 +13,7 @@
   protected @JsonView(Views.ReadOnly.class) String createdAt;
   protected @JsonView(Views.ReadOnly.class) String updatedAt;
   protected @JsonView(Views.ReadWrite.class) String name;
+  protected @JsonView(Views.ReadWrite.class) String resourceType;
```
Comment (michal-mally) [inline src/main/java/com/getbase/models/Source.java]: This one should be `ReadOnly`.

File: src/main/java/com/getbase/models/Source.java [hunk 4]
```diff
@@ -41,6 +42,14 @@ public void setName(String name) {
     this.name = name;
   }
 
+  public String getResourceType() {
+    return this.resourceType;
+  }
+
+  public void setResourceType(String resourceType) {
```
Comment (michal-mally) [inline src/main/java/com/getbase/models/Source.java]: There should be no setter for this field.

File: src/main/java/com/getbase/models/Source.java [hunk 5]
```diff
@@ -13,6 +13,7 @@
   protected @JsonView(Views.ReadOnly.class) String createdAt;
   protected @JsonView(Views.ReadOnly.class) String updatedAt;
   protected @JsonView(Views.ReadWrite.class) String name;
+  protected @JsonView(Views.ReadOnly.class) String resourceType;
```
Comment (pawelgocek) [inline src/main/java/com/getbase/models/Source.java]: This is not documented as read only. What about if some wants to create a source?

Comment (michal-mally) [reply src/main/java/com/getbase/models/Source.java]: Indeed. @ppietr will you make it `ReadWrite` again (and add setter for it)?

File: src/main/java/com/getbase/models/Deal.java [hunk 6]
```diff
@@ -32,6 +32,10 @@
   protected @JsonView(Views.ReadWrite.class) BigDecimal value;
   protected @JsonView(Views.ReadWrite.class) List<String> tags = new ArrayList<String>();
   protected @JsonView(Views.ReadWrite.class) Map<String, Object> customFields = new HashMap<String, Object>();
+  protected @JsonView(Views.ReadOnly.class) DateTime lastStageChangeAt;
```
Comment (dlwcz) [inline src/main/java/com/getbase/models/Deal.java]: last_stage_change_at is read-write now

**Comment by michal-mally** [comment]
@ppietr could you please add new fields here to have some minimal level of confidence: https://github.com/basecrm/basecrm-java/blob/master/src/test/groovy/com/getbase/services/BaseSpecification.groovy#L81

File: src/test/groovy/com/getbase/services/BaseSpecification.groovy [hunk 7]
```diff
@@ -86,6 +86,8 @@ abstract class BaseSpecification extends Specification {
                 'value'     : 1000.0,
                 'tags'      : ["important"],
                 'contact_id': createContact().id,
+                'last_stage_change_at' : "2014-09-27T16:32:56+00:00",
```
Comment (michal-mally) [inline src/test/groovy/com/getbase/services/BaseSpecification.groovy]: Please keep the format of date consistent w/ the official documentation (all 4 dates).

Comment (grzegorz-mysliwiec) [reply src/test/groovy/com/getbase/services/BaseSpecification.groovy]: Is it important for PS that the date is in format "2014-09-27T16:32:56+00:00"? 
By default API will generate it in shorter form "2014-09-27T16:32:56Z"

Comment (michal-mally) [reply src/test/groovy/com/getbase/services/BaseSpecification.groovy]: We will change it to shorter format ("2014-09-27T16:32:56Z").

**Pull request merged**